File: Java/P334.java
public class P334 {
    static java.util.Scanner entrada;
   
    public static void main (String args[]) {
        entrada = new java.util.Scanner(System.in);
        int numCasosDePrueba = Integer.parseInt(entrada.nextLine());

        for (int i = 0; i < numCasosDePrueba; i++) {
            casoDePrueba();
        }
    }

    private static void casoDePrueba(){
        String nombre = entrada.nextLine();
        
        if (nombre.length() < 3){
            nombre = "  ".concat(nombre);
        }
        
        if (esPicto(nombre)){
            if (sacarEtnia(nombre).equals("PLUS ULTRA")){
                System.out.println("PICTO");
            }else{
                System.out.println("MULATO");
            }
        }else{
            System.out.println(sacarEtnia(nombre));
        }
    }

    private static String sacarEtnia(String nombre){
        int resto = nombre.length() - 2;
        String ultimos = nombre.substring(resto, nombre.length());
        
        if ("IX".equals(ultimos)){
            return "GALO";
        } else if ("US".equals(ultimos) || "UM".equals(ultimos)) {
            return "ROMANO";
        } else if ("IC".equals(ultimos)) {
            return "GODO";
        } else if ("AS".equals(ultimos)) {
            return "GRIEGO";
        } else if ("AF".equals(ultimos)) {
            return "NORMANDO";
        } else if ("IS".equals(ultimos) || "OS".equals(ultimos) || "AX".equals(ultimos)) {
            return "BRETON";
        } else if ("EZ".equals(ultimos)) {
            return "HISPANO";
        } else if ("A".equals(ultimos.substring(1, 2))) {
            return "INDIO";
        } else {
            return "PLUS ULTRA";
        }
    }
    
    private static boolean esPicto(String nombre){
        String primeros = nombre.substring(0, 3);
        
        return primeros.equals("MAC");
    }
}

File: C++/p334.cpp
#include <cstdio>
#include <iostream>
#include <regex>
#include <string>

using namespace std;

string sacar_etnia(string nombre){
    int resto = nombre.length() - 2;
    string ultimos = nombre.substr(resto, nombre.length());

    if (ultimos == "IX"){
        return "GALO";
    } else if (ultimos == "US" || ultimos == "UM") {
        return "ROMANO";
    } else if (ultimos == "IC") {
        return "GODO";
    } else if (ultimos == "AS") {
        return "GRIEGO";
    } else if (ultimos == "AF") {
        return "NORMANDO";
    } else if (ultimos == "IS" || ultimos == "OS" || ultimos == "AX") {
        return "BRETON";
    } else if (ultimos == "EZ") {
        return "HISPANO";
    } else if (ultimos.substr(1, 2) == "A") {
        return "INDIO";
    } else {
        return "PLUS ULTRA";
    }
}

bool es_picto(string nombre){
    string primeros = nombre.substr(0, 3);
        
    return primeros == "MAC";
}

void caso_de_prueba(){
    string nombre;

    getline(cin, nombre);

    if (nombre.length() < 3){
            nombre = "   " + nombre;
    }
        
    if (es_picto(nombre)){
        if (sacar_etnia(nombre) == "PLUS ULTRA"){
            cout << "PICTO" << endl;
        }else{
            cout << "MULATO" << endl;
        }
    }else{
        cout << sacar_etnia(nombre) << endl;
    }
}

int main(){
    string cantidad;

    getline(cin, cantidad);

    for (int i = 0; i < stoi(cantidad); i++) {
        caso_de_prueba();
    }

    return 0;
}

